internal/providers/store.go

```diff
@@ -90,6 +93,57 @@ func NewProviderStore(store db.Store) ProviderStore {
 	return &providerStore{store: store}
 }
 
+func (p *providerStore) Create(ctx context.Context, projectID uuid.UUID, provParams *db.CreateProviderParams) (*db.Provider, error) {
+	if provParams == nil || projectID == uuid.Nil {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid arguments")
+	}
+
+	prov, err := p.store.CreateProvider(ctx, *provParams)
+	if err != nil {
+		return nil, fmt.Errorf("error creating provider: %w", err)
+	}
+	return &prov, nil
+}
+
+func (p *providerStore) CreateWithToken(ctx context.Context, projectID uuid.UUID, provParams *db.CreateProviderParams, tokenParams *db.UpsertAccessTokenParams) (*db.Provider, error) {
+	if provParams == nil || tokenParams == nil || projectID == uuid.Nil {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid arguments")
+	}
+
+	tx, err := p.store.BeginTransaction()
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "error starting transaction: %v", err)
+	}
+	defer p.store.Rollback(tx)
+
+	qtx := p.store.GetQuerierWithTransaction(tx)
+
+	// Check if the provider exists
+	provider, err := qtx.GetProviderByName(ctx, db.GetProviderByNameParams{
```

Comment:
this code is forklifted from what was in the GH provider service. Unsure if it's maybe too magical that CreateWithToken just silently updates the token if theprovider already exists. Maybe this could be handled outside the service, but the convenient thing about putting this method into the service was that we do have the db store and can call the transactions and I didn't want to leak them up to the provider manager which calls this method.